interfaces/portal/src/app/pages/program-payment-transaction-list/components/retry-transactions-dialog/retry-transactions-dialog.component.ts

```diff
@@ -75,30 +79,27 @@ export class RetryTransactionsDialogComponent {
         dryRun,
       }),
     onSuccess: (data, variables) => {
-      if (variables.dryRun) {
-        this.dryRunResult.set(data);
-        this.retryTransactionsConfirmationDialog().show({
-          trackingEvent: {
-            category: TrackingCategory.manageTransactions,
-            action: TrackingAction.clickProceedButton,
-            name: this.trackingEventName(),
-            value: this.trackingValue(),
-          },
+      if (data.nonApplicableCount === 0) {
+        if (variables.dryRun) {
+          this.retryFailedTransactionsMutation.mutate({ dryRun: false });
+          return;
+        }
+        this.retryTransactionsConfirmationDialog().hide();
+        this.toastService.showToast({
+          summary: $localize`Retrying transactions`,
+          detail: $localize`${data.applicableCount} transactions(s) are being retried. The status change can take up to a minute to process.`,
+          severity: 'info',
+          showSpinner: true,
         });
+
+        this.invalidateCache();
```

Comment:
ok, so functionally testing this it works all as expected, but I am a bit confused by this code. maybe explain me in a call.